### PR TITLE
Support getFileSystemContaining(path)

### PR DESCRIPTION
**Pull request by ruhan1**
Add API getFileSystemContaining(path). 
This will help Indy when trying to get a path from a group which contains thousands of repos. Iterating through those repos takes long time. With this new API, Indy can know all repos containing the target path in one db query.

### CODE REVIEW

**Review by sswguo — COMMENTED** [review]
Looks good.

File: pathdb/datastax/src/main/java/org/commonjava/storage/pathmapped/pathdb/datastax/CassandraPathDB.java [hunk 1]
```diff
@@ -542,18 +574,12 @@ public boolean copy( String fromFileSystem, String fromPath, String toFileSystem
...
         }
 
-        // check parent paths
-        String fromParentPath = PathMapUtils.getParentPath( fromPath );
         String toParentPath = PathMapUtils.getParentPath( toPath );
-        if ( fromParentPath != null && !fromParentPath.equals( toParentPath ) )
-        {
-            makeDirs( toFileSystem, toParentPath );
-        }
-
```
Comment (ligangty) [inline pathdb/datastax/src/main/java/org/commonjava/storage/pathmapped/pathdb/datastax/CassandraPathDB.java:552]: Little confused here. What's the purpose to remove this mkdir operation?

**Review by ligangty — COMMENTED** [review]
Totally looks good. One little question in comment.

File: pathdb/datastax/src/main/java/org/commonjava/storage/pathmapped/pathdb/datastax/CassandraPathDB.java (hunk 1 above)
Comment (ruhan1) [reply pathdb/datastax/src/main/java/org/commonjava/storage/pathmapped/pathdb/datastax/CassandraPathDB.java:552]: The logic is moved to insert() method. During insert(), the missing dirs will be created.

**Comment by ruhan1** [comment]
Close it. This pr has flaws. I will submit a better one soon.

**Pull request closed without merging**